Code/GraphMol/Descriptors/AUTOCORR3D.cpp

```diff
@@ -179,60 +178,63 @@ void get3DautocorrelationDesc(double* dist3D, double* topologicaldistance,
   for (unsigned int j = 0; j < 8; ++j) {
     for (unsigned int i = 0; i < 10; ++i) {
       res[j * 10 + i] =
-          round(1000 * TDBmat[j][i] / (numAtoms * (numAtoms - 1))) / 1000;
+          std::round(1000 * TDBmat[j][i] / (numAtoms * (numAtoms - 1))) / 1000;
     }
   }
 }
 
-    void get3DautocorrelationDescCustom(double* dist3D, double* topologicaldistance,
-                                  int numAtoms, const ROMol& mol,
-                                  std::vector<double>& res,
-                                  const std::string &customAtomPropName) {
-      Map<MatrixXd> dm(dist3D, numAtoms, numAtoms);
-      Map<MatrixXd> di(topologicaldistance, numAtoms, numAtoms);
-
-      std::vector<double> customAtomArray = moldata3D.GetCustomAtomProp(mol,customAtomPropName);
-      VectorXd Wc = getEigenVect(customAtomArray);
-
-      MatrixXd Bi;
-      MatrixXd tmp;
-      double TDBmat[10];
-      double dtmp;
-
-      for (int i = 0; i < 10; i++) {
-        double* Bimat = GetGeodesicMatrix(topologicaldistance, i + 1, numAtoms);
-        Map<MatrixXd> Bi(Bimat, numAtoms, numAtoms);
-        MatrixXd RBi = Bi.cwiseProduct(dm);
-
-        tmp = Wc.transpose() * RBi * Wc;
-        dtmp = (double)tmp(0);
-        if (std::isnan(dtmp)) {
-          dtmp = 0.0;
-        }
-        TDBmat[i] = dtmp;
-        delete[] Bimat;
-      }
-      // update the Output vector!
-        for (unsigned int i = 0; i < 10; ++i) {
-          res[i] = round(1000 * TDBmat[i] / (numAtoms * (numAtoms - 1))) / 1000;
-      }
+void get3DautocorrelationDescCustom(double* dist3D, double* topologicaldistance,
+                                    int numAtoms, const ROMol& mol,
+                                    std::vector<double>& res,
+                                    const std::string& customAtomPropName) {
+  Map<MatrixXd> dm(dist3D, numAtoms, numAtoms);
+  Map<MatrixXd> di(topologicaldistance, numAtoms, numAtoms);
+
+  std::vector<double> customAtomArray =
+      moldata3D.GetCustomAtomProp(mol, customAtomPropName);
+  VectorXd Wc = getEigenVect(customAtomArray);
+
+  MatrixXd Bi;
+  MatrixXd tmp;
+  double TDBmat[10];
+  double dtmp;
+
+  for (int i = 0; i < 10; i++) {
+    double* Bimat = GetGeodesicMatrix(topologicaldistance, i + 1, numAtoms);
```

Comment:
unique_ptr?

Reply:
Doing a general refactoring pass through the 3D descriptors at some point would be a really good idea, but I'd rather not get too deeply into that in this PR. I just did mechanical things (the replacement of boost::math functionality) this time through